constraintlayout/compose/src/main/java/androidx/constraintlayout/compose/MotionLayoutState.kt

```diff
@@ -0,0 +1,250 @@
+                val stateCommand = it.receive()
+
+                // Handle the command with `launch` to avoid blocking this scope, when a new Command
+                // is received and launched, Animatable will cancel any running animations from
+                // previous Commands
+                launch {
+                    when (stateCommand) {
+                        is MotionAnimationCommand.Animate -> {
+                            animatableProgress.animateTo(
+                                targetValue = stateCommand.newProgress,
+                                animationSpec = stateCommand.animationSpec
+                            )
+                        }
+                        is MotionAnimationCommand.Snap -> {
+                            animatableProgress.snapTo(targetValue = stateCommand.newProgress)
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * [MutableState] for the debug mode.
+     */
+    private val debugModeState: MutableState<MotionLayoutDebugFlags> =
+        mutableStateOf(initialDebugMode)
+
+    /**
+     * Internal observable debug mode.
+     *
+     * @see MotionLayoutDebugFlags
+     */
+    @PublishedApi
+    internal val debugMode: MotionLayoutDebugFlags
+        get() = debugModeState.value
+
+
+    // TODO: Remove once we substitute uses of `progressState` with MotionProgress
+    /**
+     * Internal [State] for the progress.
+     *
+     * Prefer to use [motionProgress] instead.
+     */
+    @PublishedApi
+    internal val progressState: State<Float> = animatableProgress.asState()
+
+    /**
+     * Object used by MotionLayout internals to read and update the progress.
+     */
+    @PublishedApi
+    internal val motionProgress: MotionProgress = object : MotionProgress {
+        override val progress: Float
+            get() = progressState.value
+
+        override suspend fun updateProgress(newProgress: Float) {
+            animatableProgress.snapTo(newProgress)
+        }
+    }
+
+    override val currentProgress: Float
+        get() = animatableProgress.value
+
+    override val isInDebugMode: Boolean
+        get() = debugModeState.value == MotionLayoutDebugFlags.SHOW_ALL
+
+    override fun setDebugMode(motionDebugFlag: MotionLayoutDebugFlags) {
+        debugModeState.value = motionDebugFlag
+    }
+
+    override fun snapTo(newProgress: Float) {
+        channel.trySend(MotionAnimationCommand.Snap(newProgress))
+    }
+
+    override fun animateTo(newProgress: Float, animationSpec: AnimationSpec<Float>) {
+        channel.trySend(MotionAnimationCommand.Animate(newProgress, animationSpec))
+    }
+}
+
+/**
+ * Returns a [MotionLayoutState], it can be used to observe and animate the progress value of a
+ * [MotionLayout] Composable.
+ *
+ * Returns the same instance if [key] is equal to the previous composition, otherwise produces and
+ * remembers a new instance.
+ */
```

Comment:
Could we add a little more about how to use this.
something like 
 val motionState = rememberMotionLayoutState(initialProgress = 0.5f, key = myKey.value)
 motionState .value ...   // to animate between x and y  etc.

Reply:
Done.